submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,304 @@
+const usersList = [];
+let usersFilteredList = [];
+const nameInput = document.querySelector('.filter__block input');
+
+const callAPI = async () => {
+    const data = await fetch('https://randomuser.me/api/?results=30&nat=ua');
+    handleErrors(data);
+
+    const { results: users } = await data.json();
+
+    return users;
+};
+
+const handleErrors = (response) => {
+    if (!response.ok) {
+        throw Error(response.statusText);
+    }
+    return response;
+}
+
+const getUsers = async () => {
+    try {
+        const users = await callAPI();
+        usersList.push(...users);
+        renderUsers(getSidebarOptions(), usersList);
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+const mainContainer = document.getElementById('main__content');
+const usersListContainer = document.createElement('div');
+usersListContainer.classList.add('users');
+
+const renderUsers = (filtersObject, usersArray) => {
+    let users = filterUsers(filtersObject, usersArray);
+    users = users.map((user) => createUser(user)).join("");
```

Comment:
This is why you should consider mostly using constants. `let`s invite mistakes if the project gets more sophisticated.

First you create a variable `users` and you place some array there. Later on, you can mistake it with some global entity `users`, since the name is too general. What actualy is happening is that you store your _filtered_ users here. So why not name it something like `filteredUsers`?

Then you _reassign_ the same variable with another value. It was a filtered array of objects, containing user informations, now it is a string, containting HTML code for the page. In the same variable named `users`. You see the trouble here?

Simple 

```js
const filteredUsers = ...
const userCards = ...
```

would massively improve readability and your future ability to quickly navigate your own code without looking for what's actually inside those variables.